datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/KafkaTopicPartitionStatsResponse.java

```diff
@@ -0,0 +1,115 @@
+/**
+ *  Copyright 2020 LinkedIn Corporation. All rights reserved.
+ *  Licensed under the BSD 2-Clause License. See the LICENSE file in the project root for license information.
+ *  See the NOTICE file in the project root for additional information regarding copyright ownership.
+ */
+package com.linkedin.datastream.connectors.kafka;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.kafka.common.TopicPartition;
+import org.codehaus.jackson.JsonNode;
+import org.codehaus.jackson.JsonParser;
+import org.codehaus.jackson.Version;
+import org.codehaus.jackson.annotate.JsonAutoDetect;
+import org.codehaus.jackson.annotate.JsonGetter;
+import org.codehaus.jackson.annotate.JsonMethod;
+import org.codehaus.jackson.annotate.JsonSetter;
+import org.codehaus.jackson.map.DeserializationConfig;
+import org.codehaus.jackson.map.DeserializationContext;
+import org.codehaus.jackson.map.JsonDeserializer;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.codehaus.jackson.map.module.SimpleModule;
+import org.codehaus.jackson.type.TypeReference;
+
+import com.linkedin.datastream.common.JsonUtils;
+
+
+/**
+ * Response structure used for Topic partition stats
+ * @see AbstractKafkaConnector#process(String)
+ */
+public class KafkaTopicPartitionStatsResponse {
+
+  private String _consumerGroupId;
+
+  private Set<TopicPartition> _topicPartitions;
+
+  /**
+   *  Default constructor for deserialization
+   */
+  public KafkaTopicPartitionStatsResponse() {
+  }
+
+  /**
+   * Constructor for KafkaTopicPartitionStatsResponse
+   * @param consumerGroupId identifier for consumer group
+   * @param topicPartitions a map of consumer offsets for topic partitions
+   */
+  public KafkaTopicPartitionStatsResponse(String consumerGroupId, Set<TopicPartition> topicPartitions) {
+    _consumerGroupId = consumerGroupId;
+    _topicPartitions = topicPartitions;
+  }
+
+  @JsonGetter("_consumerGroupId")
```

Comment:
With the auto-field discovery and the custom deserializer, could you please double-check if you need these getter/setter annotations? I have a somewhat strong suspicion we don't really need them.

I think we can do w/o `@JsonGetter`, `@JsonSetter`, and the default ctor in favor of the newer `@JsonCreator` and `@JsonProperty` annotations.
```java
  @JsonCreator
  public KafkaTopicPartitionStatsResponse(
      @JsonProperty("consumerGroupId") String consumerGroupId,
      @JsonProperty("topicPartitions") Set<TopicPartition> topicPartitions) {
    ...
  }
```